src/main/java/frc/robot/RobotContainer.java

```diff
@@ -538,6 +549,24 @@ private Command setRobotMode(RobotMode robotMode) {
         return runOnce(() -> this.robotMode = robotMode);
     }
 
+    public Pose2d getClosestBranch(Pose2d robotPose) {
+        var alliance = DriverStation.getAlliance();
+        if (alliance.isPresent() && alliance.get() == DriverStation.Alliance.Red) {
+            return robotPose.nearest(Constants.redBranchPoses);
+        } else {
+            return robotPose.nearest(Constants.blueBranchPoses);
+        }
+    }
+
+    public Pose2d getClosestCoralStationPosition(Pose2d robotPose) {
+        var alliance = DriverStation.getAlliance();
+        if (alliance.isPresent() && alliance.get() == DriverStation.Alliance.Red) {
+            return robotPose.nearest(Constants.redCoralStationPoses);
+        } else {
+            return robotPose.nearest(Constants.blueCoralStationPoses);
+        }
+    }
```

Comment:
These should be moved into the `CommandFactory` along with the commands that are being bound to the left and right bumpers. That should allow us to keep the `RobotContainer` clean.